common/views.py

```diff
@@ -28,15 +30,25 @@ def form_invalid(self, form):
 
 
 class AccountSignup(FormView):
-    form_class = UserCreationForm
+    form_class = RegisterForm
     template_name = 'common/signup.html'
     success_url = '/home'
 
     def form_valid(self, form):
         form.save()
-
         username = self.request.POST['username']
         password = self.request.POST['password1']
+
+        company_name = form.cleaned_data.get('company_name')
+        company_address = form.cleaned_data.get('company_address')
+        company_kwargs = {
+            'name': company_name,
+            'address': company_address,
+        }
+        company_form = CompanyForm(company_kwargs)
+        if company_form.is_valid():
```

Comment:
what if you get an error while creating company object?
like User is created and company is not and raises error.
You need to take care of this

Reply:
check `transaction.atomic()`